kyo-reactive-streams/shared/src/test/scala/kyo/interop/reactive-streams/StreamPublisherTest.scala

```diff
@@ -0,0 +1,49 @@
+package kyo.interop.reactivestreams
+
+import kyo.*
+import kyo.interop.reactivestreams.StreamPublisher
+import org.reactivestreams.tck.PublisherVerification
+import org.reactivestreams.tck.TestEnvironment
+import org.scalatestplus.testng.*
+
+final class StreamPublisherTest extends PublisherVerification[Int](new TestEnvironment(1000L)), TestNGSuiteLike:
```

Comment:
I took at the tck tests and it seems they check mostly for correctness without concurrency. If that's the case, I think we'll need to introduce tests since the code relies heavily on shared state